ote_sdk/ote_sdk/entities/model.py

```diff
@@ -115,6 +122,64 @@ def __init__(
         model_size_reduction: float = 0.0,
         _id: Optional[ID] = None,
     ):
+        # Initialization parameters validation
+        if TYPE_CHECKING:
+            check_parameter_str_class_name(
+                parameter=train_dataset,
+                parameter_name="train_dataset",
+                expected_class_names=["DatasetEntity", "HpoDataset", "ObjectDetectionDataset"],
+            )
```

Comment:
1. Why this check is inside branch `if TYPE_CHECKING`?

   Please, note that `typing.TYPE_CHECKING` is `True` is and only if the code is checked by mypy linter, it means that the code is NOT working indeed. So, this check will never work in real working code.

2. Also please note that this check compares the class name w.r.t. the strings that should not appear in the ote_sdk -- otherwise we will have to fix the checks in the base library (ote_sdk) when there are changes in names in some library that uses ote_sdk.

   I thought it could be sufficient to make `isinstance` with some base class -- e.g. `DatasetEntity` or some other class.
But `HpoDataset` uses its own trick -- it uses `__getattr__` method to forward call of attributes to its inner dataset field.
In this case it may require to use "duck typing" (see [article 1](https://en.wikipedia.org/wiki/Duck_typing), [article 2](https://ru.wikipedia.org/wiki/%D0%A3%D1%82%D0%B8%D0%BD%D0%B0%D1%8F_%D1%82%D0%B8%D0%BF%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F) )
So, you can create a function `is_like_dataset_entity` that checks an object using `hasattr` that it has such attributes as `__len__`, `__getitem__`, and `get_subset`.

Reply:
Thanks, checking "train_dataset" parameter attributes instead of class name